src/main/java/org/terasology/projectile/ProjectileAuthoritySystem.java

```diff
@@ -139,19 +139,20 @@ public void update(float delta) {
             Vector3f position = location.getWorldPosition();
             projectileMotion.direction = new Vector3f(projectileMotion.currentVelocity).normalize();
             HitResult result;
-            float displacement = projectileMotion.currentVelocity.length() * delta;
+            float displacement = Math.min(projectileMotion.currentVelocity.length() * delta,
+                    projectile.maxDistance - projectileMotion.distanceTravelled);
```

Comment:
So I see. Wouldn't it be possible for the distance travelled to be greater than maxDistance? In that case the displacement would be negative (since it's min)